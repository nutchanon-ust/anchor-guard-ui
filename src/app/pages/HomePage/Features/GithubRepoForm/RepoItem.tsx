import * as React from 'react';
import styled from 'styled-components/macro';
import { ReactComponent as StarIcon } from './assets/star.svg';
import { ReactComponent as NewWindowIcon } from './assets/new-window.svg';
import { A } from 'app/components/A';

interface Props {
  name: string;
  starCount: number;
  url: string;
}

export function RepoItem({ name, starCount, url }: Props) {
  return (
    <Wrapper>
      <Name>{name}</Name>
      <Info>
        <StarCount>
          <StarIcon />
          {starCount}
        </StarCount>
        <A href={url} target="_blank" rel="noopener noreferrer">
          <NewWindowIcon />
        </A>
      </Info>
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  padding: 0 1rem;
  min-height: 2.75rem;
  font-weight: 500;
  color: ${p => p.theme.text};

  &:nth-child(odd) {
    background-color: ${p => p.theme.backgroundVariant};
  }
`;

const Name = styled.div`
  flex: 1;
  padding: 0.625rem 0;
`;

const Info = styled.div`
  display: flex;
`;

const StarCount = styled.div`
  display: flex;
  align-items: center;
  min-width: 6rem;
  .icon {
    margin-right: 0.125rem;
  }
`;
