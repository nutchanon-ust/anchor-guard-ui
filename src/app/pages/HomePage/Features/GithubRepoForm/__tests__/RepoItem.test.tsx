import * as React from 'react';
import { render } from '@testing-library/react';
import { RepoItem } from '../RepoItem';

const renderRepoItem = (props: Parameters<typeof RepoItem>[number]) =>
  render(<RepoItem {...props} />);

describe('<RepoItem />', () => {
  it('should match the snapshot', () => {
    const repoItem = renderRepoItem({
      name: 'test',
      starCount: 1,
      url: 'testUrl',
    });
    expect(repoItem.container.firstChild).toMatchSnapshot();
  });

  it('should have url in <a> child', () => {
    const url = 'url1';
    const repoItem = renderRepoItem({
      name: 'test',
      starCount: 1,
      url: url,
    });
    expect(repoItem.container.querySelector('a')).toHaveAttribute('href', url);
  });

  it('should have props displayed', () => {
    const url = 'url1';
    const starCount = 1;
    const name = 'test';
    const repoItem = renderRepoItem({
      name: name,
      starCount: starCount,
      url: url,
    });
    expect(repoItem.queryByText(name)).toBeInTheDocument();
    expect(repoItem.queryByText(starCount.toString())).toBeInTheDocument();
  });
});
