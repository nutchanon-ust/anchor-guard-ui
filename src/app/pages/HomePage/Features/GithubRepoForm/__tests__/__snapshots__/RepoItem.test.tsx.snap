// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoItem /> should match the snapshot 1`] = `
.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c4:active {
  opacity: 0.4;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
  min-height: 2.75rem;
  font-weight: 500;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.625rem 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 6rem;
}

.c3 .icon {
  margin-right: 0.125rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    test
  </div>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <svg>
        star.svg
      </svg>
      1
    </div>
    <a
      class="c4"
      href="testUrl"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg>
        new-window.svg
      </svg>
    </a>
  </div>
</div>
`;
